pkg/client/client.go

```diff
@@ -881,6 +882,123 @@ func (c *SPDKClient) ReplicaRestoreStatus(replicaName string) (*spdkrpc.ReplicaR
+	}
+	return api.ProtoBackingImageToBackingImage(resp), nil
+}
+
+func (c *SPDKClient) BackingImageDelete(name, lvsUUID string) error {
+	if name == "" || lvsUUID == "" {
+		return fmt.Errorf("failed to delete SPDK backingImage: missing required parameter")
+	}
+
+	client := c.getSPDKServiceClient()
+	ctx, cancel := context.WithTimeout(context.Background(), GRPCServiceTimeout)
+	defer cancel()
+
+	_, err := client.BackingImageDelete(ctx, &spdkrpc.BackingImageDeleteRequest{
+		Name:    name,
+		LvsUuid: lvsUUID,
+	})
+	return errors.Wrapf(err, "failed to delete SPDK backing image %v", name)
+}
+
+func (c *SPDKClient) BackingImageGet(name, lvsUUID string) (*api.BackingImage, error) {
+	if name == "" || lvsUUID == "" {
+		return nil, fmt.Errorf("failed to get SPDK BackingImage: missing required parameter")
+	}
+
+	client := c.getSPDKServiceClient()
+	ctx, cancel := context.WithTimeout(context.Background(), GRPCServiceTimeout)
+	defer cancel()
+
+	resp, err := client.BackingImageGet(ctx, &spdkrpc.BackingImageGetRequest{
+		Name:    name,
+		LvsUuid: lvsUUID,
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get SPDK backing image %v", name)
+	}
+	return api.ProtoBackingImageToBackingImage(resp), nil
+}
+
+func (c *SPDKClient) BackingImageList() (map[string]*api.BackingImage, error) {
+	client := c.getSPDKServiceClient()
+	ctx, cancel := context.WithTimeout(context.Background(), GRPCServiceTimeout)
+	defer cancel()
+
+	resp, err := client.BackingImageList(ctx, &emptypb.Empty{})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list SPDK backing images")
+	}
+
+	res := map[string]*api.BackingImage{}
+	for name, backingImage := range resp.BackingImages {
+		res[name] = api.ProtoBackingImageToBackingImage(backingImage)
+	}
+	return res, nil
+}
+
+func (c *SPDKClient) BackingImageWatch(ctx context.Context) (*api.BackingImageStream, error) {
+	client := c.getSPDKServiceClient()
+	stream, err := client.BackingImageWatch(ctx, &emptypb.Empty{})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to open backing image watch stream")
+	}
+
+	return api.NewBackingImageStream(stream), nil
+}
+
+func (c *SPDKClient) BackingImageExpose(ctx context.Context, name, lvsUUID string) (exposedSnapshotLvolAddress string, err error) {
+	if name == "" || lvsUUID == "" {
+		return "", fmt.Errorf("failed to expose SPDK backing image: missing required parameter")
+	}
+
+	client := c.getSPDKServiceClient()
+	resp, err := client.BackingImageExpose(ctx, &spdkrpc.BackingImageGetRequest{
+		Name:    name,
+		LvsUuid: lvsUUID,
+	})
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to expose SPDK backing image %v in lvstore: %v", name, lvsUUID)
+	}
+	return resp.ExposedSnapshotLvolAddress, nil
+}
+
+func (c *SPDKClient) BackingImageUnexpose(ctx context.Context, name, lvsUUID string) error {
+	if name == "" || lvsUUID == "" {
+		return fmt.Errorf("failed to unexpose SPDK backing image: missing required parameter")
+	}
+
```

Comment:
Fixed.
Actually, BackingImageExpose() and BackingImageUnexpose() should generate the context in itself instead of using arguments
Just like BackingImageGet()